src/parser/mod.rs

```diff
@@ -4139,6 +4160,199 @@ impl<'a> Parser<'a> {
+    /// ```
+    pub fn parse_drop_trigger(&mut self) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
+            let trigger_name = self.parse_object_name(false)?;
+            self.expect_keyword(Keyword::ON)?;
+            let table_name = self.parse_object_name(false)?;
+            let option = self
+                .parse_one_of_keywords(&[Keyword::CASCADE, Keyword::RESTRICT])
+                .map(|keyword| match keyword {
+                    Keyword::CASCADE => ReferentialAction::Cascade,
+                    Keyword::RESTRICT => ReferentialAction::Restrict,
+                    _ => unreachable!(),
+                });
+            Ok(Statement::DropTrigger {
+                if_exists,
+                trigger_name,
+                table_name,
+                option,
+            })
+        } else {
+            self.prev_token();
+            self.expected("an object type after CREATE", self.peek_token())
+        }
+    }
+
+    pub fn parse_create_trigger(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            let name = self.parse_object_name(false)?;
+            let period = self.parse_trigger_period()?;
+
+            let events = self.parse_keyword_separated(Keyword::OR, Parser::parse_trigger_event)?;
+            let table_name = self
+                .expect_keyword(Keyword::ON)
+                .and_then(|_| self.parse_object_name(false))?;
+
+            // [ NOT DEFERRABLE | [ DEFERRABLE ] [ INITIALLY IMMEDIATE | INITIALLY DEFERRED ] ]
+            let mut deferrable: Option<bool> = None;
+
+            if self.parse_keyword(Keyword::NOT) {
+                self.expect_keyword(Keyword::DEFERRABLE)?;
+                deferrable = Some(false);
+            } else if self.parse_keyword(Keyword::DEFERRABLE) {
+                deferrable = Some(true);
+            };
+
+            let initially: Option<DeferrableInitial> = (deferrable.is_some()
+                && self.parse_keyword(Keyword::INITIALLY))
+            .then(|| {
+                Ok::<_, ParserError>(
+                    match self.expect_one_of_keywords(&[Keyword::IMMEDIATE, Keyword::DEFERRED])? {
+                        Keyword::IMMEDIATE => DeferrableInitial::Immediate,
+                        Keyword::DEFERRED => DeferrableInitial::Deferred,
+                        _ => unreachable!(),
+                    },
+                )
+            })
+            .transpose()?;
+
+            let mut referencing = vec![];
+            if self.parse_keyword(Keyword::REFERENCING) {
+                while let Some(refer) = self.parse_trigger_referencing()? {
+                    referencing.push(refer);
+                }
+            }
+
+            let (trigger_object, include_each) =
+                self.expect_keyword(Keyword::FOR).and_then(|_| {
```

Comment:
```suggestion
                self.expect_keyword(Keyword::FOR)?;
```
We can exit with an error immediately instead of nesting the logic within closures, this pattern is currently done in a few places in the PR, can we do a pass through to unnest them?

Reply:
By using the `and_then`, the logic relative to the presence or absence of a keyword is delimited in scope and I believe it makes the logic easier to follow. If I were to turn that in "the rows that follow the expect keyword", I believe it would not be as clear.

Could you please expand on the motivation for this proposed refactoring other than "we can write it also this other way"?

Reply:
Ah, yeah so to clarify with an example, semantically both following styles are roughly equivalent:

```rust
let a = doA()?;
let b = doB()?;
combine(a,b)
```
vs

```rust
doA().and_then(|a| doB().map(|b| (a,b))).and_then(|a,b| combine(a,b))
```

I think its arguable that `and_then` style is easier to follow. I can understand that its a subjective thing however.

My initial comment mentioned nesting the code which I think is one side-effect/downside of the `and_then` style because each layer of nesting in code generally tends to obfuscate the code even if slightly. (again, this is also arguable/subjective)

In any case, I did a quick search in this parser.mod file in this PR for occurrences of `and_then` and there are only 5 matches, all of which are being introduced in this PR so I think its safe to say that the codebase prefers the former style?

Reply:
Ok, I will adjust the code to the codebase style. Do let me know if you identify other analogous style divergences.

Reply:
Is it okay if I leave the `and_then` associated to `expect_one_of_keywords`?

Reply:
Now there remains only the following two `and_then`: [`parse_trigger_period`](https://github.com/LucaCappelletti94/sqlparser-rs/blob/c043b583c091dc5d1b8bf0d4cff53cc399ae20d5/src/parser/mod.rs#L4273-L4285) and [`parse_trigger_event`](https://github.com/LucaCappelletti94/sqlparser-rs/blob/c043b583c091dc5d1b8bf0d4cff53cc399ae20d5/src/parser/mod.rs#L4287-L4312). Do let me know whether you would want also these removed.

Reply:
Yes please we can update those as well thanks! Also highlighted a couple others in the latest comments I think we can update

Reply:
Done